diff --git a/node_modules/@dr.pogodin/react-native-fs/android/src/main/java/com/drpogodin/reactnativefs/ReactNativeFsModule.kt b/node_modules/@dr.pogodin/react-native-fs/android/src/main/java/com/drpogodin/reactnativefs/ReactNativeFsModule.kt
index 4e4fb5f..4656bb3 100644
--- a/node_modules/@dr.pogodin/react-native-fs/android/src/main/java/com/drpogodin/reactnativefs/ReactNativeFsModule.kt
+++ b/node_modules/@dr.pogodin/react-native-fs/android/src/main/java/com/drpogodin/reactnativefs/ReactNativeFsModule.kt
@@ -91,6 +91,36 @@ class ReactNativeFsModule(reactContext: ReactApplicationContext) :
         // NOOP
     }
 
+    fun emitOnDownloadBegin(data: WritableMap?) {
+        reactApplicationContext
+            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
+            .emit("DownloadBegin", data)
+    }
+
+    fun emitOnDownloadProgress(data: WritableMap?) {
+        reactApplicationContext
+            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
+            .emit("DownloadProgress", data)
+    }
+
+    fun emitOnDownloadResumable(data: WritableMap?) {
+        reactApplicationContext
+            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
+            .emit("DownloadResumable", data)
+    }
+
+    fun emitOnUploadBegin(data: WritableMap?) {
+        reactApplicationContext
+            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
+            .emit("UploadBegin", data)
+    }
+
+    fun emitOnUploadProgress(data: WritableMap?) {
+        reactApplicationContext
+            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
+            .emit("UploadProgress", data)
+    }
+
     @ReactMethod
     override fun appendFile(filepath: String, base64Content: String?, promise: Promise) {
         try {
diff --git a/node_modules/@dr.pogodin/react-native-fs/android/src/main/java/com/drpogodin/reactnativefs/ReactNativeFsPackage.kt b/node_modules/@dr.pogodin/react-native-fs/android/src/main/java/com/drpogodin/reactnativefs/ReactNativeFsPackage.kt
index 6cda7fd..403b57c 100644
--- a/node_modules/@dr.pogodin/react-native-fs/android/src/main/java/com/drpogodin/reactnativefs/ReactNativeFsPackage.kt
+++ b/node_modules/@dr.pogodin/react-native-fs/android/src/main/java/com/drpogodin/reactnativefs/ReactNativeFsPackage.kt
@@ -22,10 +22,10 @@ class ReactNativeFsPackage : BaseReactPackage() {
       moduleInfos[ReactNativeFsModule.NAME] = ReactModuleInfo(
         ReactNativeFsModule.NAME,
         ReactNativeFsModule.NAME,
-        canOverrideExistingModule = false,  // canOverrideExistingModule
-        needsEagerInit = false,  // needsEagerInit
-        isCxxModule = false,  // isCxxModule
-        isTurboModule = true // isTurboModule
+        false,  // canOverrideExistingModule
+        false,  // needsEagerInit
+        false,  // isCxxModule
+        true // isTurboModule
       )
       moduleInfos
     }
diff --git a/node_modules/@dr.pogodin/react-native-fs/ios/ReactNativeFs.h b/node_modules/@dr.pogodin/react-native-fs/ios/ReactNativeFs.h
index d62a899..77ea644 100644
--- a/node_modules/@dr.pogodin/react-native-fs/ios/ReactNativeFs.h
+++ b/node_modules/@dr.pogodin/react-native-fs/ios/ReactNativeFs.h
@@ -1,11 +1,25 @@
+#import <React/RCTEventEmitter.h>
+#import <React/RCTBridgeModule.h>
+
+#if __has_include(<ReactNativeFsSpec/ReactNativeFsSpec.h>)
 #import <ReactNativeFsSpec/ReactNativeFsSpec.h>
+#endif
 
-@interface ReactNativeFs : NativeReactNativeFsSpecBase <
+#if __has_include(<ReactNativeFsSpec/ReactNativeFsSpec.h>)
+@interface ReactNativeFs : RCTEventEmitter <
   NativeReactNativeFsSpec
 #if !TARGET_OS_OSX
   , UIDocumentPickerDelegate
 #endif
 >
+#else
+@interface ReactNativeFs : RCTEventEmitter <
+  RCTBridgeModule
+#if !TARGET_OS_OSX
+  , UIDocumentPickerDelegate
+#endif
+>
+#endif
 
 @property (retain) NSMutableDictionary* downloaders;
 @property (retain) NSMutableDictionary* uuids;
diff --git a/node_modules/@dr.pogodin/react-native-fs/ios/ReactNativeFs.mm b/node_modules/@dr.pogodin/react-native-fs/ios/ReactNativeFs.mm
index 1074d08..aee9e0d 100644
--- a/node_modules/@dr.pogodin/react-native-fs/ios/ReactNativeFs.mm
+++ b/node_modules/@dr.pogodin/react-native-fs/ios/ReactNativeFs.mm
@@ -580,6 +580,22 @@ RCT_EXPORT_METHOD(copyFile:(NSString *)from
     return @[@"UploadBegin",@"UploadProgress",@"DownloadBegin",@"DownloadProgress",@"DownloadResumable"];
 }
 
+- (void)emitOnDownloadBegin:(NSDictionary *)value {
+  [self sendEventWithName:@"DownloadBegin" body:value];
+}
+- (void)emitOnDownloadProgress:(NSDictionary *)value {
+  [self sendEventWithName:@"DownloadProgress" body:value];
+}
+- (void)emitOnDownloadResumable:(NSDictionary *)value {
+  [self sendEventWithName:@"DownloadResumable" body:value];
+}
+- (void)emitOnUploadBegin:(NSDictionary *)value {
+  [self sendEventWithName:@"UploadBegin" body:value];
+}
+- (void)emitOnUploadProgress:(NSDictionary *)value {
+  [self sendEventWithName:@"UploadProgress" body:value];
+}
+
 RCT_EXPORT_METHOD(downloadFile:(JS::NativeReactNativeFs::NativeDownloadFileOptionsT &)options
                   resolve:(RCTPromiseResolveBlock)resolve
                   reject:(RCTPromiseRejectBlock)reject)
@@ -1294,10 +1310,12 @@ RCT_EXPORT_METHOD(
   return NO;
 }
 
+#if __has_include(<ReactNativeFsSpec/ReactNativeFsSpec.h>)
 - (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:
     (const facebook::react::ObjCTurboModule::InitParams &)params
 {
     return std::make_shared<facebook::react::NativeReactNativeFsSpecJSI>(params);
 }
+#endif
 
 @end
diff --git a/node_modules/@dr.pogodin/react-native-fs/src/NativeReactNativeFs.ts b/node_modules/@dr.pogodin/react-native-fs/src/NativeReactNativeFs.ts
index c7a7412..3af1567 100644
--- a/node_modules/@dr.pogodin/react-native-fs/src/NativeReactNativeFs.ts
+++ b/node_modules/@dr.pogodin/react-native-fs/src/NativeReactNativeFs.ts
@@ -294,20 +294,20 @@ export interface Spec extends TurboModule {
 
   addListener(event: string): void;
 
-  readonly onDownloadBegin:
-    CodegenTypes.EventEmitter<DownloadBeginCallbackResultT>;
+  // readonly onDownloadBegin:
+  //   CodegenTypes.EventEmitter<DownloadBeginCallbackResultT>;
 
-  readonly onDownloadProgress:
-    CodegenTypes.EventEmitter<DownloadProgressCallbackResultT>;
+  // readonly onDownloadProgress:
+  //   CodegenTypes.EventEmitter<DownloadProgressCallbackResultT>;
 
-  readonly onDownloadResumable:
-    CodegenTypes.EventEmitter<DownloadResumableCallbackResultT>;
+  // readonly onDownloadResumable:
+  //   CodegenTypes.EventEmitter<DownloadResumableCallbackResultT>;
 
-  readonly onUploadBegin:
-    CodegenTypes.EventEmitter<UploadBeginCallbackArgT>;
+  // readonly onUploadBegin:
+  //   CodegenTypes.EventEmitter<UploadBeginCallbackArgT>;
 
-  readonly onUploadProgress:
-    CodegenTypes.EventEmitter<UploadProgressCallbackArgT>;
+  // readonly onUploadProgress:
+  //   CodegenTypes.EventEmitter<UploadProgressCallbackArgT>;
 
   removeListeners(count: number): void;
 
diff --git a/node_modules/@dr.pogodin/react-native-fs/src/index.ts b/node_modules/@dr.pogodin/react-native-fs/src/index.ts
index 5876456..709f241 100644
--- a/node_modules/@dr.pogodin/react-native-fs/src/index.ts
+++ b/node_modules/@dr.pogodin/react-native-fs/src/index.ts
@@ -1,5 +1,5 @@
 import { getReasonPhrase } from 'http-status-codes';
-import type { EventSubscription } from 'react-native';
+import { type EventSubscription, NativeEventEmitter } from 'react-native';
 
 import RNFS, {
   type DownloadBeginCallbackResultT,
@@ -24,6 +24,8 @@ import RNFS, {
   type UploadResultT,
 } from './NativeReactNativeFs';
 
+const fsEventEmitter = new NativeEventEmitter(RNFS as any);
+
 import {
   type EncodingT,
   type EncodingOptions,
@@ -142,7 +144,7 @@ export function downloadFile(options: DownloadFileOptionsT): {
 
   if (options.begin) {
     subscriptions.push(
-      RNFS.onDownloadBegin(res => {
+      fsEventEmitter.addListener('DownloadBegin', res => {
         if (res.jobId === jobId && options.begin) options.begin(res);
       }),
     );
@@ -150,7 +152,7 @@ export function downloadFile(options: DownloadFileOptionsT): {
 
   if (options.progress) {
     subscriptions.push(
-      RNFS.onDownloadProgress(res => {
+      fsEventEmitter.addListener('DownloadProgress', res => {
         if (res.jobId === jobId && options.progress) options.progress(res);
       }),
     );
@@ -158,7 +160,7 @@ export function downloadFile(options: DownloadFileOptionsT): {
 
   if (options.resumable) {
     subscriptions.push(
-      RNFS.onDownloadResumable(res => {
+      fsEventEmitter.addListener('DownloadResumable', res => {
         if (res.jobId === jobId && options.resumable) options.resumable(res);
       }),
     );
@@ -337,19 +339,19 @@ export function uploadFiles(options: UploadFileOptionsT): {
   }
 
   if (options.begin) {
-    subscriptions.push(RNFS.onUploadBegin(options.begin));
+    subscriptions.push(fsEventEmitter.addListener('UploadBegin', options.begin));
 
   // TODO: Deprecated, will be removed in a future release.
   } else if (options.beginCallback) {
-    subscriptions.push(RNFS.onUploadBegin(options.beginCallback));
+    subscriptions.push(fsEventEmitter.addListener('UploadBegin', options.beginCallback));
   }
 
   if (options.progress) {
-    subscriptions.push(RNFS.onUploadProgress(options.progress));
+    subscriptions.push(fsEventEmitter.addListener('UploadProgress', options.progress));
 
   // TODO: Deprecated, will be removed in a future release.
   } else if (options.progressCallback) {
-    subscriptions.push(RNFS.onUploadProgress(options.progressCallback));
+    subscriptions.push(fsEventEmitter.addListener('UploadProgress', options.progressCallback));
   }
 
   const files = options.files.map((item) => ({
